vulture_os/gui/crontab/api_clients_parser.py

```diff
@@ -70,11 +70,19 @@ def execute_parser(frontend):
                     extra={'frontend': str(frontend['name'])})
         parser.finish()
 
+def node_selected(current_node, frontend):
+    if current_node.state == "UP":
+        if frontend.node == current_node:
+            return True
+        elif current_node.is_master_mongo:
+            if not frontend.node:
+                return True
+            elif frontend.node.state != "UP":
+                return True
+    return False
```

Comment:
This function needs comments to clarify what conditions are tested

Reply:
Comments added